conf/evolutions/114-voxel-size-unit.sql

```diff
@@ -0,0 +1,16 @@
+START TRANSACTION;
+
+do $$ begin ASSERT (select schemaVersion from webknossos.releaseInformation) = 113, 'Previous schema version mismatch'; end; $$ LANGUAGE plpgsql;
+
+DROP VIEW webknossos.datasets_;
+
+CREATE TYPE webknossos.LENGTH_UNIT AS ENUM ('ym', 'zm', 'am', 'fm', 'pm', 'nm', 'µm', 'mm', 'cm', 'dm', 'm', 'hm', 'km', 'Mm', 'Gm', 'Tm', 'Pm', 'Em', 'Zm', 'Ym', 'Å', 'in', 'ft', 'yd', 'mi', 'pc');
+
+ALTER TABLE webknossos.datasets RENAME COLUMN scale TO voxelSizeFactor;
+ALTER TABLE webknossos.datasets ADD COLUMN voxelSizeUnit webknossos.LENGTH_UNIT;
```

Comment:
There shouldn’t be a need for such a migration, since the next inbox scan (every datastore does this once per minute) will report nm for all datasets that don’t have a voxel size unit in their datasource-properties.json. Then it should automatically be set for all usable datasets.

You are correct that this is nullable without a default value because unusable datasets are voxelsize-less, so a unit does not make sense for them either.